cachi2/core/package_managers/yarn_classic/main.py

```diff
@@ -43,7 +54,8 @@ def _resolve_yarn_project(project: Project, output_dir: RootedPath) -> None:
     prefetch_env = _get_prefetch_environment_variables(output_dir)
     _verify_corepack_yarn_version(project.source_dir, prefetch_env)
     _fetch_dependencies(project.source_dir, prefetch_env)
-    resolve_packages(project)
+    packages = resolve_packages(project)
+    _verify_no_offline_mirror_collisions(packages)
```

Comment:
Non-blocker, non-actionable, just a minor discussion point. I wonder if it would be even easier to follow if we adapted an approach where verifiers are transparent and return the verified object when no issues are detected. I.e. here it could be
```python
packages = _verify_no_offline_mirror_collisions(resolve_packages(project))
```
(pretty much like pydantic does it)
What I don't like about it is the way it reads (alternating direction with start in the middle of statement) and the way it will read for a longer chain of verifiers. Maybe it won't be that bad if we could have had some chaining like this:
```python
packages = resolve_packages_for(project) |> _verify_no_offline_mirror_collisions
```
?

Reply:
The `packages` will be used for generating SBOM components, so maybe I would continue on this point after this PR is merged and the focus moves to SBOM PR.

Reply:
Yes, the comment above was more to start a discussion about something that definitely out of scope for this PR.